packages/create-audius-app/tests/e2e/react.test.ts

```diff
@@ -0,0 +1,66 @@
+import { test, expect } from '@playwright/test'
+
+test('auths, fetches tracks, and favorites a track', async ({
+  page,
+  context
+}) => {
+  test.slow()
+
+  // Set entropy so we don't need to do OTP
+  const entropy = process.env.CREATE_AUDIUS_APP_TEST_ENTROPY ?? ''
+  await context.addInitScript((entropy) => {
+    if (window.location.hostname === 'audius.co') {
+      window.localStorage.setItem('hedgehog-entropy-key', entropy)
+    }
+  }, entropy)
+
+  await page.goto('localhost:4173')
+
+  const pagePromise = context.waitForEvent('page')
+  await page.getByRole('button', { name: 'Continue With Audius' }).click()
+
+  // Oauth popup
```

Comment:
i think cypress allows breaking up a big test like this into smaller tests with something like `test.step('blah blah')` so you can represent your comments into the code steps during the runner?

Reply:
I dug into this a bit to see if there was a playwright equivalent and there's not quite one, but a similar concept is technically possible; but a lot more work.

You can run a test in "serial" mode (which runs tests in dependent order) and reuse the `page` object to achieve the same concept.

Basically from this;
```tsx
  test('/signin goes to sign-in', async ({ page }) => {
    await goToPage({ page, path: 'signin' })
    await expect(
      page.getByRole('heading', { name: 'Sign Into Audius' })
    ).toBeVisible()
  })
```
to this;
```tsx
  test.describe.serial('/signin goes to sign-in', () => {
    let page
    test.beforeAll(async ({ browser }) => {
      page = await browser.newPage()
    })
    test('load up page', async () => {
      await goToPage({ page, path: 'signin' })
    })
    test('check for text', async () => {
      await expect(
        page.getByRole('heading', { name: 'Sign Into Audius' })
      ).toBeVisible()
    })
  })
```

The only weird thing is that if you're looking at the UI its not quite clear that you have to run these tests together since you could technically run a "piece" of a test at any time. 
<img width="321" alt="image" src="https://github.com/AudiusProject/audius-protocol/assets/6711655/b64e41bd-4947-41a6-8982-a7aac2d4afbb">
So if this is something we want to do more we could add some helpers that make it simpler to write and append "run above" or something to the title